src/python/pants/engine/internals/native_engine.pyi

```diff
@@ -659,11 +659,16 @@ class PyOptionId:
 class PyConfigSource:
     def __init__(self, path: str, content: bytes) -> None: ...
 
+# See src/rust/engine/src/externs/options.rs for the Rust-side versions of these types.
 T = TypeVar("T")
-# A pair of (option value, rank). See src/python/pants/option/ranked_value.py.
-OptionValue = Tuple[Optional[T], int]
-OptionListValue = Tuple[list[T], int]
```

Comment:
I got rid of the `Option{List/Dict}Value` variants. Easier to just provide `list[T]` as a type param to OptionValue. The only difference would be that `OptionValue[list[T]]` admits a None value, where `OptionListValue` did not, but so what...